codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectAggregator.java

```diff
@@ -188,12 +188,32 @@ public void operationComplete(ChannelFuture future) throws Exception {
         }
     }
 
-    private abstract static class AggregatedFullHttpMessage implements ByteBufHolder, FullHttpMessage {
+    protected AggregatedHttpRequest newAggregatedHttpRequest(HttpRequest request,
+                                                             ByteBuf content, HttpHeaders trailingHeaders) {
+        return new DefaultAggregatedHttpRequest(request, content, trailingHeaders);
+    }
+
+    protected AggregatedHttpResponse newAggregatedHttpResponse(HttpResponse response,
+                                                               ByteBuf content, HttpHeaders trailingHeaders) {
+        return new DefaultAggregatedHttpResponse(response, content, trailingHeaders);
+    }
+
+    public interface AggregatedHttpMessage extends FullHttpMessage {
+        void setTrailingHeaders(HttpHeaders trailingHeaders);
```

Comment:
`FullHttpMessage` has `trailingHeaders()` method already so you can just call `trailingHeaders().setAll()`. Could you tell me why you had to define a new interface?